### PR TITLE
[WIP] systemd integration draft.

### PR DESCRIPTION
**Pull request by zhongwencool**

### CODE REVIEW

File: active/0012-systemd-integration.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+# Integration with Systemd
+
+## Changelog
+
+* 2021-06-09: @zhongwencool Initial draft
+
+
+## Abstract
+
+Emqx currently uses a simple systemd forking type to start emqx in demon mode. The other benefits of systemd are not fully exploited, for example, systemd does not restart all emqx processes that exit abnormally (`kill GroupChildrenPid`). Systemd  just acts as a starter.
```
Comment (zmstone) [inline active/0012-systemd-integration.md:10]: daemon 
```suggestion
emqx currently uses a simple systemd forking type to start emqx in daemon mode. Systemd is only acting as a program starter, benefits of it are not fully exploited. For example, systemd does not restart all emqx processes that exit abnormally (`kill GroupChildrenPid`).
```

File: active/0012-systemd-integration.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
```
Comment (zmstone) [inline active/0012-systemd-integration.md:11]: Maybe add a summary of what is the goal of this proposal.
e.g.
```
The document is to share the idea about how we can make emqx better integrated with systemd
```

File: active/0012-systemd-integration.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+## Motivation
+
+The benefits of [systemd](https://github.com/systemd/systemd) include, but are not limited to:
+
+- The ability to configure specific system parameters for service such as `LimitNOFILE`.
```
Comment (zmstone) [inline active/0012-systemd-integration.md:16]: This is already the case even in fork mode ?
if yes, maybe better to state it here.

File: active/0012-systemd-integration.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+- The reboot interval can be configured, and restart after the machine reboots.
+- Journalctl logging can be configured.
```
Comment (zmstone) [inline active/0012-systemd-integration.md:18]: need more elaboration on this,
doe fork mode limit us from utilising journal logs ?

File: active/0012-systemd-integration.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+- Very stable, easy and configurable way to manage services.
+
+This will make the node more robust and the user will see more information about the operation of the node via `systemctl status emqx`, Make `start/stop/restart` work as expected.
```
Comment (zmstone) [inline active/0012-systemd-integration.md:21]: need more elaboration on this too about: how is `systemct` broken (or note good enough) today.

**Review by zmstone — COMMENTED** [review]
This is a great proposal, thank you @zhongwencool 

I have left some comments mostly demanding more inputs.

And a question:
what does `erlang-systemd` mean to direct start of `console` (often used when testing) and `foregroud` (in docker entrypoint)?

**Comment by zhongwencool** [comment]
OK,I am preparing to add more details.

**Pull request closed without merging**